### PR TITLE
test update media reference

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

**Comment by itkg-rboessel** [comment]
@nicolasThal

File: MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+use OpenOrchestra\MediaModelBundle\Document\Media;
+use OpenOrchestra\ModelBundle\Document\Block;
+use OpenOrchestra\ModelBundle\Document\Node;
+use OpenOrchestra\BackofficeBundle\Tests\Functional\Controller\AbstractControllerTest;
+use OpenOrchestra\ModelBundle\Document\Status;
+use OpenOrchestra\ModelInterface\Event\StatusableEvent;
+use OpenOrchestra\ModelInterface\StatusEvents;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+class UpdateMediaReferenceSubscriberTest extends AbstractControllerTest
```
Comment (nicolasThal) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: class comment

Comment (nicolasThal) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: As you perform no query, I don't think that extending AbstractControllerTest is justified

Comment (nicolasThal) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: You could only extend KernelTestCase

Comment (itkg-rboessel) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: @nicolasThal  : Client need to be authenticated to change status when it dispatch an event.

File: MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+     *
+     * @dataProvider provideMediaBlocks
+     */
+    public function testAddMediaBlocks(array $blockType, $mediaIndex)
+    {
+        $this->node->setStatus($this->pending);
+        /** @var Media $media */
+        $media = $this->medias[$mediaIndex];
+        $references = $media->getUsageReference();
+        unset($references);
```
Comment (ngilain) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: Is that usefull ? you set a variable then unset it ?

File: MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        $this->checkMediaReference($media, null);
+
+        $block = $this->generateBlock($blockType, 'ET9reyt');
+        $this->node->addBlock($block);
+
+        $attributes = $block->getAttributes();
+        $attributes['pictures'] = array(MediaInterface::MEDIA_PREFIX . $media->getId());
+        $attributes['id'] = $blockType["component"] . "Id";
+        $method = $blockType["component"] . "BlockConfiguration";
```
Comment (ngilain) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: The string "BlockConfiguration" should be declared in a class constant, as the previous "Id"

**Comment by itkg-rboessel** [comment]
@itkg-ngilain @itkg-alavieille

File: MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        $attributes['id'] = $blockType["component"] . ATTRIBUTE_ID_SUFFIX;
+        $method = $blockType["component"] . METHOD_SUFFIX;
+        $attributes = $this->$method($attributes);
+        $block->setAttributes($attributes);
+
+        $event = new StatusableEvent($this->node, $this->published);
+        $this->eventDispatcher->dispatch(StatusEvents::STATUS_CHANGE, $event);
+
+        $mediaBlockIndex = $this->node->getBlockIndex($block);
+        $expectedReference = "node-" . $this->node->getId() . "-" .  $mediaBlockIndex;
```
Comment (ngilain) [inline MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php]: this 'node' should also go into a class constant to be coherent with the other strings

**Comment by ngilain** [comment]
:+1: when the last string is turned into constant

**Pull request merged**